src/server/master_server/chunk_server_heartbeat_monitor_task.cc

```diff
@@ -0,0 +1,177 @@
+#include "src/server/master_server/chunk_server_heartbeat_monitor_task.h"
+
+#include <chrono>
+#include <memory>
+
+#include "absl/strings/str_cat.h"
+#include "src/common/protocol_client/chunk_server_control_service_client.h"
+#include "src/common/system_logger.h"
+#include "src/server/master_server/chunk_server_manager.h"
+
+using gfs::common::ConfigManager;
+using gfs::service::ChunkServerControlServiceClient;
+using google::protobuf::util::Status;
+using google::protobuf::util::StatusOr;
+using protos::grpc::CheckHeartBeatReply;
+using protos::grpc::CheckHeartBeatRequest;
+
+namespace gfs {
+
+namespace server {
+
+ChunkServerHeartBeatMonitorTask::ChunkServerHeartBeatMonitorTask()
+    : thread_(nullptr),
+      terminate_promise_(nullptr),
+      terminate_future_(nullptr),
+      config_mgr_(nullptr) {}
+
+ChunkServerHeartBeatMonitorTask::~ChunkServerHeartBeatMonitorTask() {
+  Terminate();
+}
+
+ChunkServerHeartBeatMonitorTask*
+ChunkServerHeartBeatMonitorTask::GetInstance() {
+  static ChunkServerHeartBeatMonitorTask instance;
+
+  return &instance;
+}
+
+void ChunkServerHeartBeatMonitorTask::Start(const std::string& config_file) {
```

Comment:
I think we should just pass the ConfigManager* here as we always construct this in the main, see comments below. I have extended the other service using similar pattern.

Reply:
ditto